### DISCUSSION TITLE
Elliptical nodes

### DISCUSSION THREAD

**Discussion by gcelano** [korpling/ANNIS#870] [General]
This is how an empty node could be represented according to the PAULA documentation:
```
<mark id="tok_4"
xlink:href="#xpointer(string-range(//body,'',17,3))"/><!-- out -->
<mark id="tok_5"
xlink:href="#xpointer(string-range(//body,'',21,0))"/><!-- -->
<mark id="tok_6"
xlink:href="#xpointer(string-range(//body,'',22,2))"/><!-- to --> 
```
Would it be possible (i.e., would there be problems with Annis) if I encode two elliptical nodes in the same position thus (note the indices):
```
<mark id="tok_4"
xlink:href="#xpointer(string-range(//body,'',17,3))"/><!-- out -->
<mark id="ellipsis_5"
xlink:href="#xpointer(string-range(//body,",20,0))"/><!-- -->
<mark id="ellipsis_6"
xlink:href="#xpointer(string-range(//body,'',20,0))"/><!-- -->
<mark id="tok_6"
xlink:href="#xpointer(string-range(//body,'',21,2))"/><!-- to --> 
```

**Comment by thomaskrause** [comment]
I think it would be possible in graphANNIS to represent this. But since the conversion process goes trough Pepper and thus SaltXML, you would create two `SToken` elements that have the same text coverage and you probably have to check that the PaulaXML importer supports that. I am a little worried, though, that the relANNIS exporter in Pepper might have a problem mapping two token with the same index because it models token via their text coverage

Given the example base text

| 17 | 18 | 19 | 20 | 21 | 22 |
| --- | --- | --- | --- | --- |---|
| o | u | t | | t | o |

ellipsis_5 and ellipsis_6 would both cover the same text range and for the relANNIS exporter it would not be clear which comes after the other. You could model these as spans instead which are allowed to cover the same token/text range.

**Reply by gcelano** [reply]
I have for now chosen to represent elliptical nodes with tokens such as `[0]`, `[1]`, etc., which are added inside the PAULA text. Since the PAULA text is, in any case, extracted from an original TEI XML file, and therefore the relationship with it is lost, I think that it can be accepted that the text can be modified in this sense. Annotations can then be added to `[0]` easily, as if it were a "regular" token.
